client/src/controllers/PreviewController.ts

```diff
@@ -0,0 +1,473 @@
+import { Controller } from '@hotwired/stimulus';
+import { WAGTAIL_CONFIG } from '../config/wagtailConfig';
+import { debounce } from '../utils/debounce';
+import { gettext } from '../utils/gettext';
+import type { ProgressController } from './ProgressController';
+
+/**
+ * Controls the preview panel component to submit the current form state and
+ * update the preview iframe if the form is valid.
+ */
+export class PreviewController extends Controller<HTMLElement> {
+  static classes = ['unavailable', 'hasErrors', 'selectedSize'];
+
+  static targets = [
+    'sizeInput',
+    'newTabButton',
+    'spinner',
+    'modeSelect',
+    'iframe',
+  ];
+
+  static values = {
+    baseUrl: { default: '', type: String },
+    autoUpdate: { default: true, type: Boolean },
+    autoUpdateInterval: { default: 500, type: Number },
+    deviceWidthProperty: { default: '--preview-device-width', type: String },
+    panelWidthProperty: { default: '--preview-panel-width', type: String },
+    deviceLocalStorageKey: {
+      default: 'wagtail:preview-panel-device',
+      type: String,
+    },
+  };
```

Comment:
Similar here, shorter names may help keep the html simpler and align with other controllers.


```
  static values = {
    url: { default: '', type: String }, // we use url or src in other controllers, url probably makes more sense. 
    //... Others are probably names that cannot be shorter without losing context.
  };
```